### PR TITLE
Issue #24 Show a window to select the pull secret if not configured

### PR DESCRIPTION
**Pull request by anjannath**
This will use the pull-secret as an argument to start and not save
it in crc configs

Depends on https://github.com/code-ready/crc/pull/1561

## Testing
1. Make sure `pull-secret-file` is unset in `crc`
2. Remove the already running tray and start the tray from this PR's artifact
3. Click on start, when prompted to select pull secret, click on browse and select the pull secret file
4. Follow along with `tail -f ~/.crc/crcd.log`

Screenshot 2020-10-02 at 8 24 08 AM
Screenshot 2020-10-02 at 8 24 57 AM

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ jobs:
 ## Build crc
     steps:
     - checkout
+    - run: sudo defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES
+    - run: rm ~/.ssh/id_rsa || true
+    - run: for ip in $(dig @1.1.1.1 github.com +short); do ssh-keyscan github.com,$ip; ssh-keyscan $ip; done 2>/dev/null >> ~/.ssh/known_hosts || true
```
Comment (guillaumerose) [inline .circleci/config.yml:15]: Thanks! sorry for breaking this..

File: CodeReady Containers/DaemonCommander/DaemonCommander.swift [hunk 2]
```diff
@@ -86,3 +86,19 @@ func SendCommandToDaemon(command: Request) -> Data? {
     }
     return "Failed".data(using: .utf8)
 }
+
+func SendCommandToDaemon(command: ConfigGetRequest) -> Data? {
+    do {
+        let req = try JSONEncoder().encode(command)
+        print(req)
```
Comment (guillaumerose) [inline CodeReady Containers/DaemonCommander/DaemonCommander.swift:93]: unwanted debug log ?

File: CodeReady Containers/DaemonCommander/DaemonCommander.swift [hunk 3]
```diff
@@ -86,3 +86,19 @@ func SendCommandToDaemon(command: Request) -> Data? {
...
+        let daemonConnection = DaemonCommander(sockPath: socketPath.path)
+        print(socketPath.path)
+        daemonConnection.connectToDaemon()
+        daemonConnection.sendCommand(command: req)
+        let reply = daemonConnection.readResponse()
+        return reply
```
Comment (guillaumerose) [inline CodeReady Containers/DaemonCommander/DaemonCommander.swift:99]: nit: directly `return daemonConnection.readResponse()`

File: .circleci/config.yml (hunk 1 above)
Comment (anjannath) [reply .circleci/config.yml:15]: No worries, i also didn't notice this until this PR

File: CodeReady Containers/DaemonCommander/DaemonCommander.swift (hunk 2 above)
Comment (anjannath) [reply CodeReady Containers/DaemonCommander/DaemonCommander.swift]: Done!

File: CodeReady Containers/DaemonCommander/DaemonCommander.swift (hunk 3 above)
Comment (anjannath) [reply CodeReady Containers/DaemonCommander/DaemonCommander.swift]: Done!

**Comment by anjannath** [comment]
This is missing a check for if there's an existing crc VM and not ask for the pull secret in that case.

**Pull request merged**